src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Helpers.cs

```diff
@@ -31,6 +31,7 @@ public static partial class JsonSerializer
             var reader = new Utf8JsonReader(utf8Json, isFinalBlock: true, readerState);
 
             ReadStack state = default;
+            jsonTypeInfo.Options.VerifyTypeInfoResolverIsSet();
             jsonTypeInfo.EnsureConfigured();
```

Comment:
Hmm, could the verification logic live inside `EnsureConfigured`? Or could this break other places where configuration is happening?

Reply:
I'm unsure this is safe, I think yes but currently this check is not matching all callsites